### ISSUE TITLE
Android Flavours don't build in parallel

### ISSUE THREAD

**Issue by westonal** [westonal/GradleFlavorsNonParallelExample#1]
Hi @autonomousapps. I asked a question at end of Gradle DC NYC talk on Monday.

The question was: will android ever make flavours build in parallel? You said you weren't aware they didn't already.

Here is a simple example with two flavours, not using more than one worker. This has been my experience, and often the app tasks are the longest (dex etc), so on large projects, I have seen 16 cores in parallel for 5 mins, then one thread for the next 30!

The scan is on the readme, but also I'll put it here: https://gradle.com/s/5ouqrzboqwaoy

**Comment by autonomousapps** [comment]
Thank you! Today is a holiday, but I will definitely look into this soon.
My current best guess is it's an issue in the Android Gradle Plugin;
probably even designed that way. But conceptually, I feel we should be able
to get some parallelization into that pipeline.

**Comment by autonomousapps** [comment]
**tl;dr** I blame the Android Gradle Plugin ;-)

It does in fact _look like_ it's doing more work than it needs to (but I don't purport to be an expert on that tool). It looks like it's packaging OneDebug, and then when it finishes, it packages TwoDebug. This despite the fact you have the same set of resources and source code (well, no code at all) for each flavor.

It would be interesting to see what would happen if you had flavor-specific sourceSets. I have a suspicion they would also be built in series.

Keep an eye on updates to AGP. Version 3.2 will definitely have performance enhancements, as will 3.3. I'll also raise this internally at Gradle to see if anyone knows of plans to parallelize flavors.
